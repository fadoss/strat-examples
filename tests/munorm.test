metalevel/munorm

red norm(upModule('NATS-FROM, true), 'take['s_^3['0.Zero], 'natsFrom['0.Zero]]) .
red norm('NATS-FROM, "take(3, natsFrom(0))") .
red norm('ExSec11_1_Luc02, "first(s(s(s(s(0)))), terms(s(0)))") .
