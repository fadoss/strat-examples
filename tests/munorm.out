==========================================
reduce in MAIN : norm(upModule('NATS-FROM, true), 'take['s_^3['0.Zero],
    'natsFrom['0.Zero]]) .
rewrites: 3141
result GroundTerm: '_:_['0.Zero, '_:_['s_['0.Zero], '_:_['s_^2['0.Zero],
    'nil.LazyList`{Nat`}]]]
==========================================
reduce in MAIN : norm('NATS-FROM, "take(3, natsFrom(0))") .
rewrites: 3139
result String: "0 : 1 : 2 : nil"
==========================================
reduce in MAIN : norm('ExSec11_1_Luc02, "first(s(s(s(s(0)))), terms(s(0)))") .
rewrites: 670
result String: "recip (s (0 )): recip (s (s (s (s (0 ))))): recip (s (s (s (s (s (s (s (s (s (0 )))))))))): recip (s (s (s (s (s (s (s (s (s (s (s (s (s (s (s (s (0 ))))))))))))))))): nil"
Bye.
